jina/types/arrays/neural_ops.py

```diff
@@ -82,3 +83,26 @@ def match(
                 d = Document(darray[int(_id)], copy=True)
                 d.scores[m_name] = _dist
                 _q.matches.append(d)
+
+    def visualize(
+        self, colors: Union[None, np.ndarray] = None, file_path: Union[None, str] = None
+    ):
+        """Visualize embeddings in a 2D projection with the PCA algorithm.
+
+        :param colors: Optional np.ndarray of lengh equal to len(self) containing integers for colors.
+        :param file_path: Optional path to store the visualization.
+        """
+
+        import matplotlib.pyplot as plt
+
+        pca = PCA(n_components=2)
+        x_mat = np.stack(self.get_attributes('embedding'))
+        x_mat_2d = pca.fit_transform(x_mat)
+
+        plt.scatter(x_mat_2d[:, 0], x_mat_2d[:, 1], c=colors)
+        plt.title('PCA projection')
+        plt.xlabel('PCA component 0')
+        plt.ylabel('PCA component 1')
```

Comment:
- square figure with equal axis, best fit
- please use more informative title,  e.g. '{number of data points} with PCA'
- remove x & y label, not necessary

Reply:
Done